src/interactive-window/debugger/startupCodeProvider.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { inject, injectable } from 'inversify';
+import { Uri } from 'vscode';
+import { getAssociatedNotebookDocument } from '../../kernels/helpers';
+import { IKernel, isLocalConnection, IStartupCodeProvider, StartupCodePriority } from '../../kernels/types';
+import { InteractiveWindowView } from '../../platform/common/constants';
+import { IFileSystem } from '../../platform/common/platform/types';
+import { IConfigurationService, IExtensionContext, IsWebExtension } from '../../platform/common/types';
+
+@injectable()
+export class InteractiveWindowDebuggingStartupCodeProvider implements IStartupCodeProvider {
+    public priority = StartupCodePriority.Debugging;
+    private addRunCellHookContents?: Promise<string>;
+
+    constructor(
+        @inject(IConfigurationService) private readonly configService: IConfigurationService,
+        @inject(IFileSystem) private readonly fs: IFileSystem,
+        @inject(IExtensionContext) private readonly context: IExtensionContext,
+        @inject(IsWebExtension) private readonly isWebExtension: boolean
+    ) {}
+
+    async getCode(kernel: IKernel): Promise<string[]> {
+        if (!this.isWebExtension) {
+            const useNewDebugger = this.configService.getSettings(undefined).forceIPyKernelDebugger === true;
```

Comment:
Now I merged the code for both Web and Node. I'm not sure though if the setting check on Node is still necessary but I kept it @rchiodo

Reply:
Yeah this is really confusing. Maybe a comment? The addRunCellHook is necessary when using ipykernel 6 with the old interactive window debugger.

The old interactive window debugger only supported local though.